persistence/persistence-mybatis/src/main/resources/org/cbioportal/persistence/mybatis/ClinicalDataMapper.xml

```diff
@@ -230,28 +235,55 @@
         <include refid="whereSampleSearchAndPagination"/>
     </select>
 
+    <select id="getSampleClinicalDataClinicalTableCount" resultType="Integer">
+        SELECT count(DISTINCT clinical_sample.INTERNAL_ID)
+        <include refid="fromSample"/>
+        <include refid="whereSampleSearchAndPagination"/>
+    </select>
+    
     <sql id="whereSampleSearchAndPagination">
-        INNER JOIN
-        (SELECT DISTINCT clinical_sample.INTERNAL_ID
+        INNER JOIN (
+            SELECT DISTINCT clinical_sample.INTERNAL_ID
             <if test="_parameter.containsKey('sortBy') and sortBy != null">
-                , clinical_sample.ATTR_ID, clinical_sample.ATTR_VALUE
+                <if test="sortBy == 'sampleId'">
+                  , sample.STABLE_ID AS SORT_BY
+                </if>
+                <if test="sortBy == 'patientId'">
+                  , patient.STABLE_ID AS SORT_BY
+                </if>
+                <!-- else: -->
+                <if test="sortBy != 'sampleId' and sortBy != 'patientId'">
+                  <!-- Mark samples without the sort attribute as NULL: -->
+                  , CASE 
+                        WHEN clinical_sample.ATTR_ID = #{sortBy} THEN clinical_sample.ATTR_VALUE 
+                        WHEN clinical_patient.ATTR_ID = #{sortBy} THEN clinical_patient.ATTR_VALUE 
+                        ELSE NULL 
+                    END AS SORT_BY
+                </if>
             </if>
-            <include refid="fromSample"/>
+            <include refid="fromSampleAndClinicalPatient"/>
```

Comment:
I don't quite sure why this is patient-related, the name of this mapper function is `whereSampleSearchAndPagination`. Could you elaborate on this?

Reply:
The new query also does the sorting now, possibly on patient attributes, so we need to join the clinical patient table when that happens
Any suggestions for a better name?